phoenix-core/src/main/java/org/apache/phoenix/schema/PTableImpl.java

```diff
@@ -1445,6 +1446,35 @@ public Integer getBucketNum() {
         return indexes;
     }
 
+    @Override
+    public List<PTable> getIndexes(boolean viewOnly) {
+        List<PTable> indexes = new ArrayList<>();
+        indexes.addAll(this.indexes);
+        if (viewOnly) {
+            for (PTable index: this.indexes) {
+                if(index.getTableName().toString().contains("#")) {
+                    indexes.remove(index);
+                }
+            }
+        }
+        return indexes;
+    }
+
+    @Override
+    public List<PTable> getIndexes(List<NamedNode> indexes) {
+        List<PTable> indexesPTable = Lists.newArrayListWithExpectedSize(indexes.size());
+        List<String> indexesParam = Lists.newArrayListWithExpectedSize(indexes.size());
+        for (NamedNode index : indexes) {
+            indexesParam.add(index.getName());
+        }
+        for (PTable index : this.indexes) {
+            if(indexesParam.contains(index.getTableName().getString())) {
```

Comment:
What if 'indexes' contains names that are not present?

Reply:
@swaroopak - Perhaps we should use List<String> since NamedNode is specifically part of the parse path.